spring-messaging/src/main/java/org/springframework/messaging/simp/broker/DefaultSubscriptionRegistry.java

```diff
@@ -208,304 +216,245 @@ public void unregisterAllSubscriptions(String sessionId) {
 		}
 		MultiValueMap<String, String> result = new LinkedMultiValueMap<>(allMatches.size());
 		allMatches.forEach((sessionId, subIds) -> {
-			for (String subId : subIds) {
-				SessionSubscriptionInfo info = this.subscriptionRegistry.getSubscriptions(sessionId);
-				if (info == null) {
-					continue;
-				}
-				Subscription sub = info.getSubscription(subId);
-				if (sub == null) {
-					continue;
-				}
-				Expression expression = sub.getSelectorExpression();
-				if (expression == null) {
-					result.add(sessionId, subId);
-					continue;
-				}
-				try {
-					if (Boolean.TRUE.equals(expression.getValue(messageEvalContext, message, Boolean.class))) {
-						result.add(sessionId, subId);
-					}
-				}
-				catch (SpelEvaluationException ex) {
-					if (logger.isDebugEnabled()) {
-						logger.debug("Failed to evaluate selector: " + ex.getMessage());
-					}
-				}
-				catch (Throwable ex) {
-					logger.debug("Failed to evaluate selector", ex);
-				}
+			SessionSubscriptionInfo subscriptions = this.subscriptionRegistry.getSubscriptions(sessionId);
+			if (subscriptions != null) {
+				subIds.stream()
```

Comment:
We've found in the past (especially on JDK8) that using java.util.Steam on hot code paths hinders performance. Could you change this block accordingly?

Reply:
I have never spot any observable performance degradation on JDK8 related to streams, but I will follow your advice and remove all streams for now.
I almost forgot that someone on the world is still riding that dead horse 😄